ui/dashboards/src/components/Panel/PanelHeader.tsx

```diff
@@ -35,36 +35,37 @@ export function PanelHeader({ id, title, description, editHandlers, isHovered, s
   const titleElementId = `${id}-title`;
   const descriptionTooltipId = `${id}-description`;
 
-  let action: CardHeaderProps['action'] = undefined;
+  let actions: CardHeaderProps['action'] = undefined;
   if (editHandlers !== undefined) {
     // If there are edit handlers, always just show the edit buttons
-    action = (
-      <Stack direction="row" spacing={0.5} alignItems="center">
+    actions = (
+      <>
         <InfoTooltip description="Edit">
           <HeaderIconButton aria-label={`edit panel ${title}`} size="small" onClick={editHandlers.onEditPanelClick}>
-            <PencilIcon />
+            <PencilIcon fontSize="inherit" />
```

Comment:
Forces the icon to inherit the font size from the parent button, which is set to `1.125` when `size` is `small`.